### PR TITLE
Fix get tenant context from static method

### PR DESCRIPTION
**Pull request by ayeshajay**
**Purpose**
Setting tenant context and tenant parameter in static method will replace already set value with new value for two concurrent logins in different tenant domains.

fix https://github.com/wso2/product-is/issues/11898 

Related PRs 
https://github.com/wso2-support/carbon-identity-framework/pull/1648
https://github.com/wso2-support/carbon-identity-framework/pull/1647

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 1]
```diff
@@ -3038,4 +3038,19 @@ private boolean isGroup(String localRoleName) {
...
+    /**
+     * Get tenant context using tenant domain.
+     *
+     * @param tenantDomain tenant domain of the tenant
+     * @return tenant context
+     */
+    private String getTenantContextFromTenantDomain(String tenantDomain) {
+
+        if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
+            return MultitenantConstants.TENANT_AWARE_URL_PREFIX + "/" + tenantDomain + "/";
```
Comment (ChamathNS) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:3051]: What's the value of `MultitenantConstants.TENANT_AWARE_URL_PREFIX`?

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 2]
```diff
@@ -3038,4 +3038,19 @@ private boolean isGroup(String localRoleName) {
         return !Stream.of(INTERNAL_DOMAIN, APPLICATION_DOMAIN, WORKFLOW_DOMAIN).anyMatch(domain -> localRoleName
                 .toUpperCase().startsWith((domain + UserCoreConstants.DOMAIN_SEPARATOR).toUpperCase()));
     }
+
+    /**
+     * Get tenant context using tenant domain.
+     *
+     * @param tenantDomain tenant domain of the tenant
+     * @return tenant context
```
Comment (ChamathNS) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:3046]: Let's use sentence case for param descriptions.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java (hunk 1 above)
Comment (ayeshajay) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:3051]: "t"

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java (hunk 2 above)
Comment (ayeshajay) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java]: 4c53628

**Review by ChamathNS — APPROVED** [review]

**Pull request merged**